### ISSUE TITLE
Support postgresql.UUID compilation with literal_binds=True

### ISSUE THREAD

**Issue by jmg-duarte** [sqlalchemy/sqlalchemy#7561]
### Describe the use case

In the code base I work, I have several columns using the `postgresql.UUID` type, and to validate my queries I usually compile them with `compile_kwargs={"literal_binds": True}`

### Databases / Backends / Drivers targeted

PostgreSQL (I'm using psycopg2)

### Example Use

```py
stmt.compile(dialect=psql.dialect(), compile_kwargs={"literal_binds": True})
```

Where `stmt` is any valid SQL query (i.e. `select`, `update`, etc)

### Additional context

I tried to "patch" SQLAlchemy directly in my `.venv` with the following code (and it worked):

`sqlalchemy.dialects.postgresql.base.UUID`

```py
def literal_processor(self, dialect):
        return str
```

I know this might be incorrect but I'd be more than happy to receive feedback and submit a PR that addresses this issue.

**Comment by zzzeek** [comment]
we can accept a PR for this however the string version of the value needs to successfully round-trip in PostgreSQL to work.  this is tested using the `bindparam("key", value=some_uuid, literal_execute=True)` syntax, as well as by running the compiled string into connection.execute().    tests would go at https://github.com/sqlalchemy/sqlalchemy/blob/a869dc8fe3cd579ed9bab665d215a6c3e3d8a4ca/test/dialect/postgresql/test_types.py#L2739  and for an example of how we are doing "literal binds round trip" for general types see https://github.com/sqlalchemy/sqlalchemy/blob/a869dc8fe3cd579ed9bab665d215a6c3e3d8a4ca/lib/sqlalchemy/testing/suite/test_types.py#L51

**Comment by jmg-duarte** [comment]
I'd like to be assigned this issue and move the conversation to the PR https://github.com/sqlalchemy/sqlalchemy/pull/7563

**Comment by sqla-tester** [comment]
**José Duarte** has proposed a fix for this issue in the **main** branch:

**Fixes(#7561) Add support for postgres.UUID literal_binds compilation** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3503

**Comment by sqla-tester** [comment]
**José Duarte** has proposed a fix for this issue in the **rel_1_4** branch:

**Fixes(#7561) Add support for postgres.UUID literal_binds compilation** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3523

**Comment by simonquartz32-lang** [comment]
1
